### PR TITLE
Implement Schema update

### PR DESCRIPTION
**Pull request by tschmidtb51**
- resolves secvisogram/secvisogram#24
- implement current version
- add new schemas
- fix min and max documents
- fix examples
- fix tests

### CODE REVIEW

**Comment by tschmidtb51** [comment]
Please see the corresponding [oasis-tcs/csaf](https://github.com/oasis-tcs/csaf) as mentioned in #24.

**Comment by tschmidtb51** [comment]
@sthagen, @pdamian: I added the mssing documentation and fixed the preview.

File: app/lib/SecvisogramPage/View/FormEditorTab/Document/Publisher.js [hunk 1]
```diff
@@ -36,14 +50,14 @@ export default React.memo(
             <TextAttribute
               {...publisherProps('issuing_authority')}
               label="Issuing authority"
-              description="The name of the issuing party and their authority to release the document, in particular, the party's constituency and responsibilities or other obligations."
+              description="Provides information about the authority of the issuing party to release the document, in particular, the party's constituency and responsibilities or other obligations.."
```
Comment (pdamian) [inline app/lib/SecvisogramPage/View/FormEditorTab/Document/Publisher.js:53]: Two dots ".." at the end.

**Review by pdamian — APPROVED** [review]
Identified only a minor issue which could potentially be improved.

If you come up with (standard-conform) acknowledgements like this...
`
    "acknowledgments": [
      {
        "names": [
          "Johann"
        ]
      },
      {
        "organization": "CISA"
      },
      {
        "summary": "First analysis"
      },
      {
        "urls": [
          "https://"
        ]
      }
    ]
`
...this could end in "not so nice looking" previews. Especially for **acknowledgments having only an organization** show up in an empty bullet:

We thank the following parties for their efforts: 
- Johann
- 
- for First analysis
- (see: https://)

File: app/lib/SecvisogramPage/View/FormEditorTab/Document/Publisher.js (hunk 1 above)
Comment (tschmidtb51) [reply app/lib/SecvisogramPage/View/FormEditorTab/Document/Publisher.js]: Fixed with 7aee81c.

**Comment by tschmidtb51** [comment]
> Identified only a minor issue which could potentially be improved.
> 
> If you come up with (standard-conform) acknowledgements like this...
> `"acknowledgments": [ { "names": [ "Johann" ] }, { "organization": "CISA" }, { "summary": "First analysis" }, { "urls": [ "https://" ] } ]`
> ...this could end in "not so nice looking" previews. Especially for **acknowledgments having only an organization** show up in an empty bullet:
> 
>> We thank the following parties for their efforts:
>> 
>>  * Johann
>>  *
>>  * for First analysis
>>  * (see: https://)

I wasn't able to reproduce that. Did you do a `npm ci` and `npm run dev` on you local machine or did you use a online tool? My machine shows for your example the following result (The structure was changed with this PR):

>CSAF TC Tester thanks the following parties for their efforts:
>
>- Johann
>- CISA
>- for First analysis
>-  (see: https://)

This is the valid representation for the given structure. We now explain in the [standard](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.0/prose/csaf-v2-editor-draft.md#3115-acknowledgments-type---example) that the structure:
```
"acknowledgments": [
      {
        "names": [
          "Johann"
        ],
        "organization": "CISA",
        "summary": "First analysis",
        "urls": [
          "https://"
        ]
      }
    ]
```

should provide the outcome:

>CSAF TC Tester thanks the following parties for their efforts:
>- Johann from CISA for First analysis (see: https://)

Do you think that this is enough guidance from the standard?

**Comment by pdamian** [comment]
> > Identified only a minor issue which could potentially be improved.
> > If you come up with (standard-conform) acknowledgements like this...
> > `"acknowledgments": [ { "names": [ "Johann" ] }, { "organization": "CISA" }, { "summary": "First analysis" }, { "urls": [ "https://" ] } ]`
> > ...this could end in "not so nice looking" previews. Especially for **acknowledgments having only an organization** show up in an empty bullet:
> > > We thank the following parties for their efforts:
> > > 
> > > * Johann
> > > 
> > > * for First analysis
> > > * (see: https://)
> 
> I wasn't able to reproduce that. Did you do a `npm ci` and `npm run dev` on you local machine or did you use a online tool?

Yes, I did a `npm ci` and `npm run dev` on my local machine.

> My machine shows for your example the following result (The structure was changed with this PR):
> 
> > CSAF TC Tester thanks the following parties for their efforts:
> > 
> > * Johann
> > * CISA
> > * for First analysis
> > * (see: https://)
> 
> This is the valid representation for the given structure. We now explain in the [standard](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.0/prose/csaf-v2-editor-draft.md#3115-acknowledgments-type---example) that the structure:
> 
> ```
> "acknowledgments": [
>       {
>         "names": [
>           "Johann"
>         ],
>         "organization": "CISA",
>         "summary": "First analysis",
>         "urls": [
>           "https://"
>         ]
>       }
>     ]
> ```
> 
> should provide the outcome:
> 
> > CSAF TC Tester thanks the following parties for their efforts:
> > 
> > * Johann from CISA for First analysis (see: https://)
> 
> Do you think that this is enough guidance from the standard?

I did a new checkout of the latest commits, and indeed the problem disappeared (I guess my initial checkout this morning missed some of your commits later on during the day).

**Comment by pdamian** [comment]
By the way, is there a way to easily change the listening port when using `npm run dev`? This is helpful if port 8080 is already in use, or if for development you want to run two instances of Secvisogram in parallel. If there is such a way, it would be nice to have this documented in the README.md/Getting Started section.

**Comment by tschmidtb51** [comment]
> I did a new checkout of the latest commits, and indeed the problem disappeared (I guess my initial checkout this morning missed some of your commits later on during the day).

Sorry about the confusion. I realized after I pushed the first commits that I was missing the preview...

**Comment by tschmidtb51** [comment]
> By the way, is there a way to easily change the listening port when using `npm run dev`? This is helpful if port 8080 is already in use, or if for development you want to run two instances of Secvisogram in parallel. If there is such a way, it would be nice to have this documented in the README.md/Getting Started section.

Not that I know of. Feel free to investigate ;-)

**Comment by pdamian** [comment]
> > By the way, is there a way to easily change the listening port when using `npm run dev`? This is helpful if port 8080 is already in use, or if for development you want to run two instances of Secvisogram in parallel. If there is such a way, it would be nice to have this documented in the README.md/Getting Started section.
> 
> Not that I know of. Feel free to investigate ;-)

Here you go: https://github.com/secvisogram/secvisogram/pull/26

**Review by sthagen — APPROVED** [review]
LGTM

**Pull request merged**